internal/dcache/debug/debug_fs.go

```diff
@@ -0,0 +1,164 @@
+   |     |  |      |     | |     | |     |     | |       |            |
+   | --- |  |      |     | |-----| |---- |     | |-----| |-----  ------
+   |     |  |      |     | |     | |     |     |       | |       |
+   | ____|  |_____ | ____| | ____| |     |_____|  _____| |_____  |_____
+
+
+   Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+
+   Copyright © 2020-2025 Microsoft Corporation. All rights reserved.
+   Author : <blobfusedev@microsoft.com>
+
+   Permission is hereby granted, free of charge, to any person obtaining a copy
+   of this software and associated documentation files (the "Software"), to deal
+   in the Software without restriction, including without limitation the rights
+   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+   copies of the Software, and to permit persons to whom the Software is
+   furnished to do so, subject to the following conditions:
+
+   The above copyright notice and this permission notice shall be included in all
+   copies or substantial portions of the Software.
+
+   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+   SOFTWARE
+*/
+
+package debug
+
+import (
+	"fmt"
+	"io"
+	"sync"
+	"syscall"
+
+	"github.com/Azure/azure-storage-fuse/v2/common"
+	"github.com/Azure/azure-storage-fuse/v2/internal"
+	"github.com/Azure/azure-storage-fuse/v2/internal/handlemap"
+)
+
+// This package gives the debug facility to the dcache. Users can use top level sub-directory as "fs=debug" to know the
+// state of the cluster/cache(maybe regarding clusterinfo, rpc calls, etc...). The files this package serves would be
+// created on the fly and not stored anywhere in the filesystem.
+
+var procFiles map[string]*procFile
+
+// Mutex, openCnt is used for correctness of the filesystem if more that one handles for these procFiles are opened.
+// without which also one can implement given that always there would one handle open for the file.
+type procFile struct {
+	mu            sync.Mutex            // lock for updating the openCnt and refreshing the buffer.
+	buf           []byte                // Contents of the file.
+	openCnt       int                   // Open handles for this file.
+	refreshBuffer func(*procFile) error // Refresh the contents of the file.
+}
+
+// Directory entries in "fs=debug" directory. This list don't change as the files we support were already known.
+var procDirList []*internal.ObjAttr
+
+func init() {
+	// Register the callbacks for the procFiles.
+	procFiles = map[string]*procFile{
+		"clusterMap.json": &procFile{
```

Comment:
let's not use extension and camelcase in files/dirs in debug.
so call this "clustermap"